pkgs/development/libraries/gegl/reproducibility_fix.patch

```diff
@@ -0,0 +1,26 @@
+--- a/gegl/gegl-config.c	2023-03-26 14:19:28.000000000 +0100
++++ b/gegl/gegl-config.c	2023-05-07 13:18:51.971220727 +0100
+@@ -322,13 +322,7 @@
+     mem_available = (uint64_t) sysconf (_SC_AVPHYS_PAGES) * sysconf (_SC_PAGESIZE);
+ #endif
+ 
+-    default_tile_cache_size = mem_total;
+-    if (default_tile_cache_size > mem_available)
+-    {
+-      default_tile_cache_size = mem_available;
+-    }
+-    if (default_tile_cache_size < mem_min)
+-      default_tile_cache_size = mem_min;
++    default_tile_cache_size = 1024 * 1024 * 1024;
```

Comment:
The same holds for this property.

Reply:
Actually, it looks like this is just initialized from `GeglBufferConfig` so all of this work is for nothing and it probably gets always initialized to `512 * 1024 * 1024`:

https://gitlab.gnome.org/GNOME/gegl/-/blob/401c6d30b82e557dc3d8fa12979d3b0ecb6a0463/gegl/gegl-config.c#L436-438
https://gitlab.gnome.org/GNOME/gegl/-/blob/401c6d30b82e557dc3d8fa12979d3b0ecb6a0463/gegl/buffer/gegl-buffer-config.c#L166

Reply:
Yeah, verified that the value is just whatever `GeglBufferConfig` is set too:

```patch
diff --git a/pkgs/development/libraries/gegl/default.nix b/pkgs/development/libraries/gegl/default.nix
index bd4aea3210f..e19b535e091 100644
--- a/pkgs/development/libraries/gegl/default.nix
+++ b/pkgs/development/libraries/gegl/default.nix
@@ -111,10 +111,12 @@ stdenv.mkDerivation rec {
   postPatch = ''
     chmod +x tests/opencl/opencl_test.sh
     patchShebangs tests/ff-load-save/tests_ff_load_save.sh tests/opencl/opencl_test.sh tools/xml_insert.sh
+
+    substituteInPlace gegl/buffer/gegl-buffer-config.c --replace '512 * 1024 * 1024' '512'
   '';
 
   # tests fail to connect to the com.apple.fonts daemon in sandboxed mode
-  doCheck = !stdenv.isDarwin;
+  doCheck = false;
 
   meta = with lib; {
     description = "Graph-based image processing framework";
```

```ShellSession
$ nix-shell -I nixpkgs=$HOME/Projects/nixpkgs -p glib gegl gobject-introspection 'python3.withPackages(ps: with ps; [ pygobject3 ])' --run 'python3 -c "from gi.repository import Gegl; c = Gegl.Config(); print(c.props.tile_cache_size)"'
<string>:1: PyGIWarning: Gegl was imported without specifying a version first. Use gi.require_version('Gegl', '0.4') before import to ensure that the right version gets loaded.
512
```

So turns out this whole patch does not do anything: https://gitlab.gnome.org/GNOME/gegl/-/commit/be8cd9eff3b82177123a39b545b9c6f36fdf6190

Reply:
would we be better off modifying the output with sed or an xml editor instead of modifying the source?

Reply:
If it is not compiled into the typelib, that might work. Though I still think using a fixed value in the `ParamSpec` and then initializing it to the correct value in `gegl_config_init` function might be more upstreamable.